test/screens/admin/edit_choice_board_category_test.dart

```diff
@@ -0,0 +1,134 @@
+      expect(find.byKey(const ValueKey("instructionsText")), findsOneWidget);
+      expect(find.byKey(const ValueKey("pickImageFromGallery")), findsOneWidget);
+      expect(find.byKey(const ValueKey("takeImageWithCamera")), findsOneWidget);
+      expect(find.byKey(const ValueKey("categoryNameField")), findsOneWidget);
+      expect(find.byKey(const ValueKey("editCategoryButton")), findsOneWidget);
+    });
+  });
+
+  testWidgets("making no edits takes user to choice board page", (WidgetTester tester) async {
+    mockNetworkImagesFor(() async {
+      await tester.pumpWidget(ThemeProvider(
+          themeNotifier: CustomTheme(),
+          child: MaterialApp(
+              home: EditChoiceBoardCategory(
+            categoryId: breakfastCategory.id,
+            categoryImageUrl: "food.jpeg",
+            categoryName: "food.jpeg",
+            auth: mockAuth,
+            firestore: mockFirestore,
+            storage: mockStorage,
+          ))));
+
+      await createData();
+
+      await tester.tap(find.byKey(ValueKey("editCategoryButton")));
+      await tester.pumpAndSettle();
+      expect(find.byType(AdminChoiceBoards), findsOneWidget);
+    });
+  });
+
+  testWidgets("changing only name is successful", (WidgetTester tester) async {
+    mockNetworkImagesFor(() async {
+      await tester.pumpWidget(ThemeProvider(
+          themeNotifier: CustomTheme(),
+          child: MaterialApp(
+              home: EditChoiceBoardCategory(
+            categoryId: breakfastCategory.id,
+            categoryImageUrl: "food.jpeg",
+            categoryName: "food.jpeg",
+            auth: mockAuth,
+            firestore: mockFirestore,
+            storage: mockStorage,
+          ))));
+
+      await createData();
+
+      final nameField = find.byKey(ValueKey("categoryNameField"));
+      await tester.enterText(nameField, "Lunch");
+      await tester.pumpAndSettle();
+
+      await tester.tap(find.byKey(ValueKey("editCategoryButton")));
+      await tester.pumpAndSettle();
+      expect(find.byType(AdminChoiceBoards), findsOneWidget);
+    });
+  });
+
+  testWidgets("editing a category that doesn't exist shows error", (WidgetTester tester) async {
+    mockNetworkImagesFor(() async {
+      await tester.pumpWidget(ThemeProvider(
+          themeNotifier: CustomTheme(),
+          child: MaterialApp(
+              home: EditChoiceBoardCategory(
+            categoryId: breakfastCategory.id,
+            categoryImageUrl: "food.jpeg",
+            categoryName: "food.jpeg",
+            auth: mockAuth,
+            firestore: mockFirestore,
+            storage: mockStorage,
+          ))));
+
+      final nameField = find.byKey(ValueKey("categoryNameField"));
+      await tester.enterText(nameField, "Dinner");
+      await tester.pumpAndSettle();
+
+      await tester.tap(find.byKey(ValueKey("editCategoryButton")));
+      await tester.pumpAndSettle();
+      expect(find.byType(ScaffoldMessenger), findsOneWidget);
```

Comment:
AlertDialog is correct but this is because I didn't use the `ScaffoldMessenger.of(context).showSnackBar( SnackBar(content: Text("message")), );` code that you recommended. I changed this because of the errors I was getting when executing your tests as I had pointed out the other day, though I can change it back

Reply:
I'm not sure why you get errors but having ScaffoldMessenger in the expect doesn't test what's intended